source/retryable-writes/retryable-writes.rst

```diff
@@ -381,36 +382,35 @@ writes, drivers MUST add a transaction ID to the command. Drivers MUST only
 attempt to retry a write command if the first attempt yields a retryable error.
 Drivers MUST NOT attempt to retry a write command on any other error.
 
-If the first attempt of a write command including a transaction ID encounters a
-retryable error, the driver MUST update its topology according to the SDAM spec
-(see: `Error Handling`_) and capture this original retryable error. Drivers
-should then proceed with selecting a writable server for the retry attempt.
+If the first attempt of a write command including a transaction ID encounters
+a retryable error, the driver MUST update its topology according to the SDAM
+spec (see: `Error Handling`_) and capture this original retryable error.
 
-.. _Error Handling: ../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#error-handling
+Drivers MUST then retry the operation as many times as necessary until any one
+of the following conditions is reached:
+
+- the operation succeeds.
+- the operation fails with a non-retryable error.
+- CSOT is enabled and the operation times out per `Client Side
+  Operations Timeout: Retryability
+  <../client-side-operations-timeout/client-side-operations-timeout.rst#retryability>`__.
+- CSOT is not enabled and one retry was attempted.
 
-If the driver cannot select a server for the retry attempt or the selected
-server does not support retryable writes, retrying is not possible and drivers
-MUST raise the original retryable error. In both cases, the caller is able to
-infer that an attempt was made.
+For each retry attempt, drivers MUST select a writable server. If the driver
+cannot select a server for a retry attempt or the selected server does not
+support retryable writes, retrying is not possible and drivers MUST raise the
+retryable error from the previous attempt. In both cases, the caller is able
+to infer that an attempt was made.
 
-If the retry attempt also fails, drivers MUST update their topology according to
+If a retry attempt also fails, drivers MUST update their topology according to
 the SDAM spec (see: `Error Handling`_). If an error would not allow the caller
 to infer that an attempt was made (e.g. connection pool exception originating
-from the driver), the original error should be raised. If the retry failed due
-to another retryable error or some other error originating from the server, that
-error should be raised instead as the caller can infer that an attempt was made
-and the second error is likely more relevant (with respect to the current
-topology state).
+from the driver) or the error is labeled "NoWritesPerformed", the error from
```

Comment:
Is there any other case besides `NoWritesPerformed` where we return the previous error?

Reply:
Yes, when a retry selects a server that does not support retryable writes or the retry fails with a server selection error.